hyperspy/model.py

```diff
@@ -2107,3 +2217,80 @@ def __getitem__(self, value):
                     "\" not found in model")
         else:
             return list.__getitem__(self, value)
+
+
+class ModelSpecialSlicers:
+
+    def __init__(self, model, isNavigation):
+        self.isNavigation = isNavigation
+        self.model = model
+
+    def __getitem__(self, slices):
+        array_slices = self.model.spectrum._get_array_slices(
+            slices,
+            self.isNavigation)
+        _spectrum = self.model.spectrum._slicer(slices, self.isNavigation)
+        if _spectrum.metadata.Signal.signal_type == 'EELS':
+            _model = _spectrum.create_model(
+                auto_background=False,
+                auto_add_edges=False)
+        else:
+            _model = _spectrum.create_model()
+
+        dims = self.model.axes_manager.navigation_dimension, self.model.axes_manager.signal_dimension
+        if self.isNavigation:
+            _model.channel_switches[:] = self.model.channel_switches
+        else:
+            _model.channel_switches[:] = \
+                np.atleast_1d(
+                    self.model.channel_switches[tuple(array_slices[-dims[1]:])])
+        from hyperspy import components
+        for _ in xrange(len(_model)):
+            _model.remove(0)
+        twin_dict = {}
+        for comp in self.model:
+            init_args = {}
+            for k, v in comp._whitelist.iteritems():
+                if v is None:
+                    continue
+                flags_str, value = v
+                if 'init' in parse_flag_string(flags_str):
+                    init_args[k] = value
+            _model.append(getattr(components, comp._id_name)(**init_args))
+
+        # TODO: create sliceable whitelist? Deal with low_loss slicing, etc.
+        copy_slice_from_whitelist(
+            self.model,
+            _model,
+            dims,
+            (slices, array_slices),
+            self.isNavigation)
+        for co, cn in zip(self.model, _model):
+            copy_slice_from_whitelist(co,
+                                      cn,
+                                      dims,
+                                      (slices, array_slices),
+                                      self.isNavigation)
+            for po, pn in zip(co.parameters, cn.parameters):
+                copy_slice_from_whitelist(po,
+                                          pn,
+                                          dims,
+                                          (slices, array_slices),
+                                          self.isNavigation)
+                twin_dict[id(po)] = ([id(i) for i in list(po._twins)], pn)
+
+        for k in twin_dict.keys():
+            for tw_id in twin_dict[k][0]:
+                twin_dict[tw_id][1].twin = twin_dict[k][1]
+
+        _model.chisq.data = _model.chisq.data.copy()
+        _model.dof.data = _model.dof.data.copy()
+        if not self.isNavigation:
```

Comment:
Looks like it is still necessary (fails tests without it). Currently trying to figure out why I need it to begin with....